File: aspnet-core/src/HRMv2.Core/Manager/ChartDetails/Dto/CreateChartDetailDto.cs
﻿using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using HRMv2.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static HRMv2.Constants.Enum.HRMEnum;

namespace HRMv2.Manager.ChartDetails.Dto
{
    [AutoMap(typeof(ChartDetail))]
    public class CreateChartDetailDto 
    {
        public long ChartId { get; set; }

        public string Name { get; set; }

        public string Color { get; set; }

        public List<long> JobPositionIds { get; set; }

        public List<long> LevelIds { get; set; }

        public List<long> BranchIds { get; set; }

        public List<long> TeamIds { get; set; }

        public List<UserType> UserTypes { get; set; }

        public List<PayslipDetailType> PayslipDetailTypes { get; set; }

        public List<EmployeeStatus> WorkingStatuses { get; set; }
    }
}

File: angular/src/app/service/model/chart-settings/chart-detail-settings/create-chart-detail.dto.ts
export class CreateChartDetailDto {
    chartId: number;
    name: string;
    color: string;
    branchIds: number[];
    jobPositionIds: number[];
    levelIds: number[];
    teamIds: number[];
    payslipDetailTypes: number[]; // app enum ESalaryType
    userTypes: number[]; // app enum usertype
    workingStatuses: number[]; // app enum UserStatus
    gender: number[];
  }
  